source/Nevermore/RelationalTransaction.cs

```diff
@@ -116,22 +117,18 @@ public class RelationalTransaction : IRelationalTransaction
         [Pure]
         public IEnumerable<T> LoadStream<T>(IEnumerable<string> ids) where T : class, IId
         {
-            var blocks = ids
-                .Distinct()
-                .Select((id, index) => (id: id, index: index))
-                .GroupBy(x => x.index / 500, y => y.id)
-                .ToArray();
-
-            foreach (var block in blocks)
-            {
-                var results = TableQuery<T>()
-                    .Where("[Id] IN @ids")
-                    .Parameter("ids", block.ToArray())
-                    .Stream();
-
-                foreach (var result in results)
-                    yield return result;
-            }
+            var idList = ids.Where(id => !string.IsNullOrWhiteSpace(id)).Distinct().ToList();
+            if (idList.Count == 0)
+                return new List<T>();
+            
+            var mapping = mappings.Get(typeof(T));
+            var tableName = mapping.TableName;
+            
+            var param = new CommandParameterValues();
+            param.AddTable("criteriaTable", idList);
+            return ExecuteReader<T>(
+                $"SELECT s.* FROM dbo.[{tableName}] s INNER JOIN @criteriaTable t on t.[ParameterValue] = s.[Id] order by s.[Id]",
```

Comment:
The `ORDER BY` caught my eye, I can't think of a horrible side effect but thought it worth calling out.

Reply:
Will drop it